### DISCUSSION TITLE
"Sorry, this server doesn't do :textDocument/declaration" with nimlsp

### DISCUSSION THREAD

**Discussion by halloleo** [joaotavora/eglot#705] [Q&A] [answered]
Trying out eglot with `nimlsp` (for [Nim](https://nim-lang.org)). Would be great to have an _easy-to-use_ lsp client when hacking away Nim code. :)

When I am on a function call (of a function properly defined in the same file) I would like to jump to the declartion of the function. So I issue `eglot-find-declaration`.

However  I then get the following error "_Sorry, this server doesn't do :textDocument/declaration_" - Why's that? And any ideas how I can fix this?

Thanks for any pointers!

PS: When I used `lsp-mode` with the same `nimlsp` it does do the jump to the declaration as expected.

**Comment by cabins** [comment]
I got the same error with jdtls (Eclipse Java language server) , while lsp-mode works as expected too.

![image](https://user-images.githubusercontent.com/2500123/155057610-18367ebd-276c-41cb-8d52-558b983cc3fe.png)

**Reply by nemethf** [reply]
Logs you attached to #836 show that your server doesn't say it a declationProvider, that's why you see the message.

**Reply by cabins** [reply]
Thank you so much!! As you mentioned at [842](https://github.com/joaotavora/eglot/discussions/842#discussioncomment-2238959), I found that only the jar file can not jump as expected, the .java scene works well.

Thank you again!!

**Reply by nemethf** [reply]
definition != declaration

**Comment by nemethf** [comment, marked as answer]
> "_Sorry, this server doesn't do :textDocument/declaration_" - Why's that? And any ideas how I can fix this?

This means the server said it had no support for the textDocument/declaration method in the initial exchange between the client and the server.  If you want to fix this, you need to modify the server.  More likely though, you can use find-definition, which more servers seem to support.  The [README](https://github.com/joaotavora/eglot#find-definition) has more info about this.

**Reply by halloleo** [reply]
Thanks for the clarification between declaration and definition. Makes sense

Also, the ReadMe is really great now with all the use cases! :-)
